src/services/user/hooks/publicTeachers.js

```diff
@@ -18,8 +21,29 @@ const filterForPublicTeacher = (hook) => {
 	// Limit accessible user (only teacher which are discoverable)
 	hook.params.query.roles = ['teacher'];
 
-	if (process.env.IGNORE_DISCOVERABILITY !== 'true') {
-		hook.params.query.discoverable = true;
+	const TEACHER_VISIBILITY_FOR_EXTERNAL_TEAM_INVITATION =	hook.app.Config
+		.get('TEACHER_VISIBILITY_FOR_EXTERNAL_TEAM_INVITATION');
+	switch (TEACHER_VISIBILITY_FOR_EXTERNAL_TEAM_INVITATION) {
+		case 'opt-in':
+			// own school and other schools if opted-in
+			hook.params.query.$or = [
+				{ schoolId: hook.params.account.schoolId },
+				{ discoverable: true },
+			];
+			break;
+		case 'opt-out':
+			// everybody who did not opt-out
+			hook.params.query.discoverable = { $ne: false }; // must not be false
+			break;
+		case 'disabled':
+			// force foreign school id to be set
+			if (equalIds(hook.params.account.schoolId, hook.params.query.schoolId) !== true) {
+				throw new Forbidden('access to foreign school teachers prohibited');
```

Comment:
should this really throw or return an empty list instead?

Reply:
throw, however this depends on configuration because of this it should fail